ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_utils.cpp

```diff
@@ -156,33 +209,124 @@ OpConfig::OpConfig(BinaryOpType binary_op_type) {
+            } else {
+                TT_THROW("Unsupported binary op for FPU {}", binary_op_type);
+            }
+            break;
+        case BinaryOpType::BITWISE_XOR:
+            if (is_sfpu_op) {
+                binary_op = SfpuBinaryOp::BITWISE_XOR;
+            } else {
+                TT_THROW("Unsupported binary op for FPU {}", binary_op_type);
+            }
+            break;
+        case BinaryOpType::LEFT_SHIFT:
+            if (is_sfpu_op) {
+                binary_op = SfpuBinaryOp::LEFT_SHIFT;
+            } else {
+                TT_THROW("Unsupported binary op for FPU {}", binary_op_type);
+            }
+            break;
+        case BinaryOpType::RIGHT_SHIFT:
+            if (is_sfpu_op) {
+                binary_op = SfpuBinaryOp::RIGHT_SHIFT;
+            } else {
+                TT_THROW("Unsupported binary op for FPU {}", binary_op_type);
+            }
+            break;
+        case BinaryOpType::POWER:
+            if (is_sfpu_op) {
+                binary_op = SfpuBinaryOp::POWER;
+            } else {
+                TT_THROW("Unsupported binary op for FPU {}", binary_op_type);
+            }
+            break;
         default: TT_THROW("Unsupported binary op {}", binary_op_type);
     }
 }
 
-std::map<std::string, std::string> OpConfig::as_defines() const {
-    std::map<std::string, std::string> defines;
+std::pair<std::string, std::string> OpConfig::get_sfpu_init_fn(DataType dtype) const {
+    if (std::holds_alternative<SfpuBinaryOp>(binary_op)) {
+        auto sfpu_binary_op = std::get<SfpuBinaryOp>(binary_op);
+        switch (sfpu_binary_op) {
+            case SfpuBinaryOp::ADD:
+                if (dtype == DataType::INT32) {
+                    return {"add_int32_tile_init();", "add_int32_tile"};
+                } else {
+                    return {"add_binary_tile_init();", "add_binary_tile"};
+                }
+            case SfpuBinaryOp::SUB: return {"sub_binary_tile_init();", "sub_binary_tile"};
+            case SfpuBinaryOp::MUL: return {"mul_binary_tile_init();", "mul_binary_tile"};
+            case SfpuBinaryOp::DIV: return {"div_binary_tile_init();", "div_binary_tile"};
+            case SfpuBinaryOp::POWER: return {"power_binary_tile_init();", "power_binary_tile"};
+            case SfpuBinaryOp::RSUB: return {"rsub_binary_tile_init();", "rsub_binary_tile"};
+            case SfpuBinaryOp::LEFT_SHIFT: return {"binary_shift_tile_init();", "binary_left_shift_tile"};
+            case SfpuBinaryOp::RIGHT_SHIFT: return {"binary_shift_tile_init();", "binary_right_shift_tile"};
+            case SfpuBinaryOp::BITWISE_AND: return {"binary_bitwise_tile_init();", "and_binary_tile"};
+            case SfpuBinaryOp::BITWISE_OR: return {"binary_bitwise_tile_init();", "or_binary_tile"};
+            case SfpuBinaryOp::BITWISE_XOR: return {"binary_bitwise_tile_init();", "xor_binary_tile"};
+            default: TT_THROW("Unsupported sfpu binary op {}", binary_op); return {"", ""};
```

Comment:
`TT_THROW` invokes a `[[noreturn]]` function so the `return` afterwards is dead code. Should just be:
```suggestion
            default: TT_THROW("Unsupported sfpu binary op {}", binary_op);
```